File: 2016/9. Explosives in Cyberspace/1.Solution.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ifstream input("input.txt");
    char ch;
    int ans = 0;
    while (input.get(ch)) {
        if (ch == '(') {
            string num1;
            string num2;
            while (input.get(ch) && ch != 'x') num1.push_back(ch);
            while (input.get(ch) && ch != ')') num2.push_back(ch);
            int togo = stoi(num1);
            ans += togo * stoi(num2);
            while (togo--) input.get(ch);
        } else ans++;
    }
    cout << ans;
}

File: 2017/5. A Maze of Twisty Trampolines, All Alike/1.Solution.py
j = []

for i in range(1056): # no of lines of input
    j.append(int(input()))

ans = 0

ind = 0
while ind < len(j) and ind >= 0:
    j[ind] += 1
    ind = ind + j[ind] - 1
    ans += 1

print(ans)
